### ISSUE TITLE
Required Problem Parameters for flow-over-heated-plate

### ISSUE THREAD

**Issue by HishamSaeed** [precice/tutorials#226]
Hi i am trying to couple openFOAM with Elmer FEM solver, i am using the flow over heated plate problem as a test problem. For this tutorial, i cannot find the density of the fluid as well as the solid which are required by Elmer solver, otherwise the solver stops and does not continue. I cannot calculate the fluid density from Prandtl number because the density cancels out in equation 
```$$Pr = \nu / \alpha $$ , $$Pr = (\mu / \rho) / (\cp * k / \rho) $$``` .
For more details about Prandtl number equation: https://en.wikipedia.org/wiki/Prandtl_number

Also in the solid i cannot calculate it because `alpha(thermal diffusivity)` is provided as well as `k(Heat Conductivity`) but the `cp(Heat Capacity)` is not provided which also is required by Elmer. Is there a way to access these data or calculate it.

Thanks in advance
Hisham

**Comment by DavidSCN** [comment]
According to OpenFOAM, the fluid density has initially a uniform value of `1 kg/m^3` (and of course varies during the course of the simulation). AFAIU you are running the solid participant, right? Does a density value of `1 kg/m^3` work for you?

**Comment by BenjaminRodenberg** [comment]
> According to OpenFOAM, the fluid density has initially a uniform value of `1 kg/m^3` (and of course varies during the course of the simulation).

Then this information should also go into our documentation. I will open a PR on this.

**Comment by BenjaminRodenberg** [comment]
> Does a density value of `1 kg/m^3` work for you?

From my perspective the density should not be required for solving the heat equation in the solid, if you are only interested in heat flux and temperature. In this case [thermal diffusivity](https://en.wikipedia.org/wiki/Thermal_diffusivity) and [thermal conductivity](https://en.wikipedia.org/wiki/Thermal_conductivity) should be sufficient. So our current specification for the solid is complete from this perspective.

However, as soon as you are interested in the actual energy you either need the [specific heat capacity](https://en.wikipedia.org/wiki/Specific_heat_capacity) or the density. Additionally, we do not know what different solvers require. Even if the [FEniCS-based heat solver](https://github.com/precice/tutorials/blob/1672e89181e66346127ba40cd15cb513d3fafb39/flow-over-heated-plate/solid-fenics/solid.py) works without density, other solvers may compute the energy (even if we don't need it) and therefore the density will be required.

My opinion here: If we also provide a density for the solid, we are on the safe side and can assume that with most solvers one will be able to set up the case.

What I would still like to know: What does Elmer require for heat conduction in a solid? What about OpenFOAM? Is there a default value? Is density not required at all? Any input would be appreciated.

**Comment by BenjaminRodenberg** [comment]
> Also in the solid i cannot calculate it because `alpha(thermal diffusivity)` is provided as well as `k(Heat Conductivity`) but the `cp(Heat Capacity)` is not provided which also is required by Elmer. Is there a way to access these data or calculate it.

From this I understand that:

* Elmer is the solid. It requires the specific heat capacity (cp) and therefore the user has to know density or specific heat capacity of the solid. This is currently not given in our documentation.
* OpenFOAM is the fluid. As described by David, OpenFOAM defaults to rho = 1 kg/m^3, but this is not documented for our tutorial case. So not clearly specified.

@HishamSaeed Is this correct or am I misunderstanding something?

**Comment by HishamSaeed** [comment]
> > Also in the solid i cannot calculate it because `alpha(thermal diffusivity)` is provided as well as `k(Heat Conductivity`) but the `cp(Heat Capacity)` is not provided which also is required by Elmer. Is there a way to access these data or calculate it.
> 
> From this I understand that:
> 
> * Elmer is the solid. It requires the specific heat capacity (cp) and therefore the user has to know density or specific heat capacity of the solid. This is currently not given in our documentation.
> * OpenFOAM is the fluid. As described by David, OpenFOAM defaults to rho = 1 kg/m^3, but this is not documented for our tutorial case. So not clearly specified.
> 
> @HishamSaeed Is this correct or am I misunderstanding something?

In my case, yes Elmer is the solid for the flow over heated plate, and Elmer always requires the density and heat capacity for the heat equation, i am not sure why. check the heat equation model in Models manual http://www.nic.funet.fi/pub/sci/physics/elmer/doc/ElmerModelsManual.pdf.

Possibly because it operates on the governing equation of incompressible  fluid.

**Comment by HishamSaeed** [comment]
> According to OpenFOAM, the fluid density has initially a uniform value of `1 kg/m^3` (and of course varies during the course of the simulation). AFAIU you are running the solid participant, right? Does a density value of `1 kg/m^3` work for you?

Yes, i am running the solid participant as Elmer, the density values for both the solid and fluid are 1kg/m^3. But i still have problems in the calculated velocity when both solvers are coupled.

**Comment by DavidSCN** [comment]
> But i still have problems in the calculated velocity when both solvers are coupled.

Could you explain this a bit more in detail?

**Comment by BenjaminRodenberg** [comment]
> > According to OpenFOAM, the fluid density has initially a uniform value of `1 kg/m^3` (and of course varies during the course of the simulation). AFAIU you are running the solid participant, right? Does a density value of `1 kg/m^3` work for you?
> 
> But this is not defined directly, right? If it was, it should be in the [constant](https://github.com/precice/tutorials/tree/master/flow-over-heated-plate/fluid-openfoam/constant) directory.

What about this part?

https://github.com/precice/tutorials/blob/6d90ae3eb179113beb400e472c8832e29a48db9b/flow-over-heated-plate/fluid-openfoam/constant/thermophysicalProperties#L31-L35

Maybe the density is "hidden" here. But I'm just guessing here and did not find a reference linking molar weight and `nMoles` (what does this mean?) to density.

**Comment by MakisH** [comment]
Exactly, one can use the molecular information in a state equation and compute e.g. the density. OpenFOAM does have state equations, here we set the `p*V=nMoles*R*T` (perfect gas):

```
equationOfState perfectGas;
```

One can usually then compute the density as:

```
nMoles / V = p / (R T),   rho = mass / V = nMoles * molWeight / V 
```

**Comment by BenjaminRodenberg** [comment]
I did a quick experiment with `molWeight 24.0999` (bottom) vs. `molWeight 24000.0999` (top) to find out whether it is actually a "don't care" or not:

![Screenshot from 2021-08-11 17-37-37](https://user-images.githubusercontent.com/5740604/129059898-5bd5890d-8cac-4914-a81d-e02a374480d4.png)

There is a clearly visible difference.

**Comment by BenjaminRodenberg** [comment]
I think I found the correct connection between the fluid density and the molar mass. I've written down everything in latex for nicer typesetting and will provide it here as a reference:

![Screenshot from 2021-08-12 08-10-48](https://user-images.githubusercontent.com/5740604/129146798-1de08d88-06b8-4dfd-ae44-9eb2ab05e6f9.png)

**Comment by DavidSCN** [comment]
Thanks, should be resolved from now on.

**Comment by DavidSCN** [comment]
For the future: a nice overview of all parameters is also given in Lucias thesis on page 53
